InnerEye/ML/SSL/datamodules_and_datasets/datamodules.py

```diff
@@ -140,19 +154,21 @@ def get_combined_loader(self, encoder_loader: Sized, linear_head_loader: Sized)
         :param encoder_loader: The dataloader to use for the SSL encoder.
         :param linear_head_loader: The dataloader to use for the linear head.
         """
-        mode = "max_size_cycle" if len(encoder_loader) > len(linear_head_loader) else "min_size"
+        mode = self._cycle_mode(len(encoder_loader), len(linear_head_loader))
         dataloaders = {
             SSLDataModuleType.ENCODER: encoder_loader,
             SSLDataModuleType.LINEAR_HEAD: linear_head_loader
         }
         return CombinedLoader(dataloaders, mode=mode)
 
-    def train_dataloader(self, *args: Any, **kwargs: Any) -> CombinedLoader:  # type: ignore
+    def train_dataloader(self, *args: Any, **kwargs: Any) -> Dict[SSLDataModuleType, DataLoader]:  # type: ignore
         """
         The train dataloaders
         """
-        return self.get_combined_loader(encoder_loader=self.encoder_module.train_dataloader(),
-                                        linear_head_loader=self.linear_head_module.train_dataloader())
+        return {
```

Comment:
Minor: I would add also here a comment about the bug to explain why we return a dict instead of using the combined_datalaoder